### DISCUSSION TITLE
Field value reference in validation

### DISCUSSION THREAD

**Discussion by rtritto** [elysiajs/elysia#846] [Q&A]
With 2 query parameters allowed, at least 1 query parameter must be provided (required).

E.g.:

OK
- ?q1=v1&q2=v2
- ?q1=v1
- ?q2=v2

KO
- `<NOTHING>`
- ?q3=v3

Draft:
```ts
import { t } from 'elysia'

app.get('/', ({ query }) => {
  ...
}, {
  query: t.Object({
    q1: t.Optional(t.String(), /* true if q2 is provided (e.g. use q2 reference for condition) */),
    q2: t.Optional(t.String(), /* true if q1 is provided (e.g. use q1 reference for condition) */)
  })
})
```

**Comment by mtt-artis** [comment]
Hi 
I've opened a ticket because the validation schema works, but it's causing a TypeScript error.
#848

**Comment by rtritto** [comment]
Opened a discussion on TypeBox: https://github.com/sinclairzx81/typebox/discussions/1004
